src/app/render/simple_renderer.h

```diff
@@ -24,6 +24,7 @@ namespace app {
     void setRefLayersVisiblity(const bool visible) override;
     void setNonactiveLayersOpacity(const int opacity) override;
     void setNewBlendMethod(const bool newBlend) override;
+    void setComposeGroups(const bool composeGroups) override;
```

Comment:
This declaration is missing in the ShaderRender, getting this compilation error:

```
/usr/bin/ccache /usr/bin/clang++ -DCLIP_ENABLE_IMAGE=1 -DCMARK_STATIC_DEFINE -DCURL_STATICLIB -DDEBUGMODE -DENABLE_DATA_RECOVERY -DENABLE_DEVMODE -DENABLE_NEWS -DENABLE_SAVE -DENABLE_SCRIPTING -DENABLE_UI -DENABLE_UPDATER -DENABLE_WEBP -DENABLE_WEBSOCKET -DHAVE_CONFIG_H -DHAVE_CONFIG_OVERRIDE_H=1 -DIXWEBSOCKET_USE_ZLIB -DLAF_LINUX -DLAF_SKIA -DLAF_WITH_CLIP -DLAF_WITH_REGION -DLIBARCHIVE_STATIC -DLUA_FLOORN2I=F2Ifloor -DLUA_USE_LINUX=1 -DPNG_NO_MMX_CODE -DSK_ALLOW_STATIC_GLOBAL_INITIALIZERS=1 -DSK_BUILD_FOR_UNIX -DSK_ENABLE_SKSL=1 -DSK_GAMMA_APPLY_TO_A8 -DSK_GAMMA_SRGB -DSK_GL=1 -DSK_INTERNAL -DSK_R32_SHIFT=16 -DSK_SCALAR_TO_FLOAT_EXCLUDED -DSK_SUPPORT_GPU=1 -DTINYXML2_DEBUG -D_DEBUG -D_FILE_OFFSET_BITS=64 -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/zlib -I/home/david/builds/aseprite-gcc/third_party/zlib -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/libpng -I/home/david/builds/aseprite-gcc/third_party/libpng -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/tinyxml2 -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/pixman/pixman -I/home/david/builds/aseprite-gcc -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/giflib -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/jpeg -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/curl/include -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/simpleini -I/home/david/deps/skia/third_party/externals/libwebp/src -I/home/david/Dropbox/ASEPRITE/aseprite/src/observable -I/home/david/builds/aseprite-gcc/src/app -I/home/david/Dropbox/ASEPRITE/aseprite/laf -I/home/david/builds/aseprite-gcc/laf -I/home/david/Dropbox/ASEPRITE/aseprite/src -I/home/david/Dropbox/ASEPRITE/aseprite/src/doc/.. -I/home/david/deps/skia -I/home/david/deps/skia/third_party/externals/freetype/include -I/home/david/deps/skia/third_party/externals/harfbuzz/src -I/home/david/Dropbox/ASEPRITE/aseprite/src/fixmath/.. -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/cityhash/src -I/home/david/Dropbox/ASEPRITE/aseprite/src/dio/.. -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/fmt/include -I/home/david/Dropbox/ASEPRITE/aseprite/src/observable/. -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/cmark/src -I/home/david/builds/aseprite-gcc/third_party/cmark/src -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/giflib/lib -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/json11 -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/libarchive/libarchive -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/tinyexpr -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/qoi -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/lua -I/home/david/Dropbox/ASEPRITE/aseprite/third_party/IXWebSocket -Wall -Wno-switch -Wimplicit-fallthrough -g -fsanitize=address   -Wno-sign-compare -std=c++17 -MD -MT src/app/CMakeFiles/app-lib.dir/ui/editor/editor_render.cpp.o -MF src/app/CMakeFiles/app-lib.dir/ui/editor/editor_render.cpp.o.d -o src/app/CMakeFiles/app-lib.dir/ui/editor/editor_render.cpp.o -c /home/david/Dropbox/ASEPRITE/aseprite/src/app/ui/editor/editor_render.cpp
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/app/ui/editor/editor_render.cpp:12:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/app/ui/editor/editor_render.h:12:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/app/render/renderer.h:11:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/render/render.h:15:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/doc/doc.h:14:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/doc/brush.h:15:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/doc/image_ref.h:12:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/doc/image.h:14:
In file included from /home/david/Dropbox/ASEPRITE/aseprite/src/doc/image_buffer.h:19:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/memory:75:
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unique_ptr.h:1065:34: error: allocating an object of abstract class type 'app::ShaderRenderer'
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                                 ^
/home/david/Dropbox/ASEPRITE/aseprite/src/app/ui/editor/editor_render.cpp:50:23: note: in instantiation of function template specialization 'std::make_unique<app::ShaderRenderer>' requested here
    m_renderer = std::make_unique<ShaderRenderer>();
                      ^
/home/david/Dropbox/ASEPRITE/aseprite/src/app/render/renderer.h:64:18: note: unimplemented pure virtual method 'setComposeGroups' in 'ShaderRenderer'
    virtual void setComposeGroups(const bool composeGroups) = 0;
                 ^
1 error generated.
```

Reply:
This problem was solved here: https://github.com/aseprite/aseprite/pull/4507/files#diff-170a556a8307dc589d017fecbfffc96294038cf70e5c23f4d15a3b9ffe37298a